linera-execution/src/runtime.rs

```diff
@@ -368,7 +390,19 @@ where
             .await
             .get(&self.application_id())
         {
-            Some(view) => Ok(view.get(&key).await?),
+            Some(view) => {
+                let result = view.get(&key).await?;
+                self.num_reads.fetch_add(1, Ordering::Relaxed);
+                if let Some(value) = &result {
+                    self.bytes_read
+                        .fetch_add(value.len() as u64, Ordering::Relaxed);
+                }
+                let bytes_read = self.bytes_read.load(Ordering::Acquire);
+                if bytes_read > self.maximum_bytes_read {
+                    return Err(ExecutionError::ExcessiveRead);
+                }
+                Ok(result)
+            }
             None => Err(ExecutionError::ApplicationStateNotLocked),
```

Comment:
Since the `Some` arm is now quite long, you could use `ok_or_else` (or `ok_or` if execution errors are small) instead of the `match`. (Also below.)

Reply:
I tried that and could not find a solution easily. Since this has no functional impact, I propose not do this change.

Reply:
Doesn't `.ok_or_else(|| ExecutionError::ApplicationStateNotLocked)?` work?

Reply:
No.

Reply:
I could find a way to address the livedness issues.